### PR TITLE
submissions: add preprint_date to form data

### PR DESCRIPTION
**Pull request by harunurhan**
* INSPIR-2806

### CODE REVIEW

File: backend/inspirehep/submissions/marshmallow/literature.py [hunk 1]
```diff
@@ -163,6 +165,7 @@ def build_literature(self, data):
         literature.add_document_type(data["document_type"])
 
         literature.add_arxiv_eprint(data.get("arxiv_id"), data.get("arxiv_categories"))
+        literature.add_preprint_date(data.get("preprint_date"))
```
Comment (drjova) [inline backend/inspirehep/submissions/marshmallow/literature.py]: add a test for the ``None`` case

Comment (harunurhan) [reply backend/inspirehep/submissions/marshmallow/literature.py]: all other unit tests expect `test_load_preprint_date`,  are already testing when this is `None`.
